packages/app/src/cli/services/app-logs/poll-app-logs.test.ts

```diff
@@ -148,17 +182,40 @@ describe('pollAppLogs', () => {
       apiKey: API_KEY,
       stdout,
     })
-
-    expect(stdout.write).toHaveBeenCalledWith('Function executed successfully using 0.5124M instructions.')
-    expect(stdout.write).toHaveBeenCalledWith(expect.stringContaining(LOGS))
-    expect(stdout.write).toHaveBeenCalledWith(expect.stringContaining('Log: '))
+    expect(writeAppLogsToFile).toHaveBeenCalledWith({
+      appLog: RESPONSE_DATA.app_logs[2],
+      apiKey: API_KEY,
+      stdout,
+    })
+    expect(writeAppLogsToFile).toHaveBeenCalledWith({
+      appLog: RESPONSE_DATA.app_logs[3],
+      apiKey: API_KEY,
+      stdout,
+    })
 
     expect(components.useConcurrentOutputContext).toHaveBeenCalledWith(
       {outputPrefix: SOURCE, stripAnsi: false},
       expect.any(Function),
     )
 
+    // app_logs[0]
+    expect(stdout.write).toHaveBeenCalledWith('Function executed successfully using 0.5124M instructions.')
+    expect(stdout.write).toHaveBeenCalledWith(expect.stringContaining(LOGS))
+    expect(stdout.write).toHaveBeenCalledWith(expect.stringContaining('Log: '))
+
+    // app_logs[1]
+    expect(stdout.write).toHaveBeenCalledWith(`❌ Function failed to execute with error: ${FUNCTION_ERROR}`)
+    expect(stdout.write).toHaveBeenCalledWith(expect.stringContaining(LOGS))
+    expect(stdout.write).toHaveBeenCalledWith(expect.stringContaining('Log: '))
+
+    // app_logs[2]
+    expect(stdout.write).toHaveBeenCalledWith('Function executed successfully using 0.5124M instructions.')
+    expect(stdout.write).toHaveBeenCalledWith(expect.stringContaining(LOGS))
+    expect(stdout.write).toHaveBeenCalledWith(expect.stringContaining('Log: '))
```

Comment:
I don't think these assertions check the call order so this one would pass even if only the `app_logs[0]` output was printed once.

You could probably use [this](https://jestjs.io/docs/expect#tohavebeennthcalledwithnthcall-arg1-arg2-) or [this](https://jestjs.io/docs/expect#tohavebeencalledtimesnumber) to validate that it was called multiple times as you expected.

Reply:
So I could see that the spy was invoked 22 times, 11 times per API invocation. and I am quite confident that the above code checks the first 11 spy invocations in order.

Reply:
done